### PR TITLE
Add post-signup and post-login overrides for redirects

### PR DESCRIPTION
**Pull request by andrew-propelauth**

### CODE REVIEW

File: src/client/AuthProvider.tsx [hunk 1]
```diff
@@ -196,8 +198,20 @@ export const AuthProvider = (props: AuthProviderProps) => {
...
+        } else {
+            return "/api/auth/login"
+        }
+    }
+    const getSignupPageUrl = (opts?: RedirectToSignupOptions) => {
+        if (opts?.postSignupRedirectPath) {
+            return `/api/auth/signup?return_to_path=${encodeURIComponent(opts.postSignupRedirectPath)}`
+        } else {
+            return "/api/auth/signup"
+        }
```
Comment (itailevi98) [inline src/client/AuthProvider.tsx:213]: (nitpick)

We can do early returns here:

```ts
if (...) {
    return ...;
}

return ...;
```

File: src/server/app-router.ts [hunk 2]
```diff
@@ -165,16 +177,21 @@ export function getRouteHandlers(args?: RouteHandlerArgs) {
...
-            const path = args?.postLoginRedirectPathFn ? args.postLoginRedirectPathFn(req) : "/"
-            if (!path) {
-                console.log("postLoginPathFn returned undefined")
+
+            // If we have a return_to_path cookie, we'll use that
+            // Otherwise, we'll use the postLoginRedirectPathFn
+            const returnToPathFromCookie = req.cookies.get(RETURN_TO_PATH_COOKIE_NAME)?.value;
+            const returnToPath = returnToPathFromCookie ?? (args?.postLoginRedirectPathFn ? args.postLoginRedirectPathFn(req) : "/")
+            if (!returnToPath) {
+                console.log("postLoginRedirectPathFn returned undefined")
```
Comment (itailevi98) [inline src/server/app-router.ts:186]: Do we want to log here in production? I saw in most of the other frontend repos there is an eslint rule to error on console.log, just want to make sure this is meant to be here

**Review by itailevi98 — APPROVED** [review]
Lgtm, just left some minor comments :)

File: src/server/app-router.ts (hunk 2 above)
Comment (andrew-propelauth) [reply src/server/app-router.ts:186]: This should probably be console.error, that's a good point. I do want to give a message to the user, but log is probably wrong, I'll fix it.

**Pull request merged**